### DISCUSSION TITLE
New bot listening in and broadcasting

### DISCUSSION THREAD

**Discussion by JacquesFreud** [jamulussoftware/jamulus#3032] [General]
Hi folks,
Since yesterday, March 23 2023, we came across a new function on the website Jamulus.live
It is called "Listen" and enables people from outside to hear what is going on in a Jamulus server, without actually joining in. After fooling around with it for a bit, some of us started to raise questions. Feel free to read along with the mail I got from one of my fellow jammers, translated from German:

_Yesterday I checked this "obscure" Jamulus.Live website with my Norton 360 security program - and was immediately warned about this "unsafe" website: Everything you enter there is tapped and distributed in a completely uncontrolled manner. My suspicion that the bot that "listened" to St. Aposteln (Note from JacquesFreud: the particular server we used) is spreading everything we say and play there - whether we like it or not - directly onto the WWW and is thus doing illegal "broadcasting", unfortunately confirmed. As nice and simple as this monitoring and recording function may seem - it violates all personal and copyright laws.
It is actually only a matter of time before major companies such as Sony or Universal Music, with the help of AI, which is already searching the www 24/7 for all possible copyright infringements, tell us about the IP addresses  and hold them liable as the perpetrators of illegal broadcasting. In Germany, the following applies: "Ignorance does not protect against punishment". And unfortunately I had to learn to see these connections only from a professional perspective.

Actually, we shouldn't stay so naively around the musical campfire any longer and continue covering songs that we don't have the rights to "publish" if these bots are broadcasting us worldwide. Unfortunately, the very good basic idea of Jamulus is destroyed by these bots. The founding fathers of Jamulus expressly give no guarantee in the "Help Menu" and refer to the GNU General Public License under US law.
That's why I rehearse with my bands via Jamulus exclusively via my private server, which you can only access with the current IP address in the "Server address" connect field. This is a closed group that no longer appears in the Jamulus directories - but all loud distortion guitarists and bad bots stay outside. 
Another possibility is the private, password-protected connection platform - Pear to Pear - on Sonobus, which also offers better sound and single track recording. However, my studio PC only manages 6-7 participants there with an acceptable latency.

You can of course continue as before - but then don't be surprised about possibly very expensive legal warning procedures in the future._

So, what do we have to say about this?

**Comment by pljones** [comment]
It's always been the way since day one.  If you're running or using a _public server_, you do whatever you do there in public - just as public as performing the same act on the town hall steps.  Any passer-by is free to watch or record you.

**Reply by JacquesFreud** [reply]
That is a clear answer. And an acceptable one, I would think. Thanks.

**Reply by rdica** [reply]
In all actuality, if you operate the server, you're free to use any tools you desire to control access to your server, regardless that jamulus software itself has no access control mechanism.

Firewalling on the server is an effective means of access control, and is used by many server operators.

**Comment by ghost** [comment]
This is more than illegal broadcasting, it is spying. People who join a Jamulus session are part of a conversation where everyone is conversing. Bots (and people) that connect without the knowledge of the group conversing are spying on that group. This is Criminal and can be prosecuted.

**Comment by chrisrimple** [comment]
Full disclosure: I have no relationship to or interest in Jamulus.live

Settle down, people.  As noted above, "public" Jamulus servers are just that: public.  Anyone can join, including a bot that is then resharing the content from the server in other ways.  That's not illegal broadcasting or spying, since the bot connects to the server using a Jamulus client just like any musician, and all musicians on the server can see that the bot has connected.

In regard to IP violations and takedown requests from music rights holders, there is zero chance that happens.  No one is going to get sued or arrested for copyright infringement.  The WorldJam has been live-streaming Jamulus sessions of musicians playing cover songs for 3 years via YouTube, which are also made available afterward as recordings.  That's almost 100 broadcasts and more than 1000 cover songs, with maybe 5 takedown alerts from YouTube in all that time.  After responding to the alert, the recording was allowed to remain on YouTube, because the recordings ARE NOT BEING MONETIZED.  That is, the musicians are not making any money from advertising or selling the music without a license.  Musicians playing on a public Jamulus server, that perhaps gets re-streamed by a bot, are not making money from the broadcast and no music rights owner is going to come after them.  Yes, the music rights owner might have a right to do so, but it's not worth their time and effort.

**Comment by ElectricBoogalo** [comment]
@mcfnord As this is you hosting this on Jamulus.live, can you please give us your reply? It is getting to the point where people are not wanting to play, and move completely private. I am feeling like as a good service you could take these "bots" out, and let things be as they used to. You probably don't' agree, as its your project. But at the very least if someone has a server and requests to have this "technology add on" removed, would you oblige to the request? That seems like the moral and kind thing to do. Thanks!

**Reply by mcfnord** [reply]
I do maintain [an opt-out list of IP:PORT values](https://jamulus.live/cannot-dock.txt), and also avoid servers that describe themselves as private. It's also reasonable to block the three IPs listed below. So of course any server operator can opt out easily. I've also been thinking about granularity at the user level, which would give individual users the power to opt out of streaming and sampling.

Streaming has helped me stay in touch with communities where I'm active. I often use the feature while riding my bicycle. I stop to send chats to the musicians. Others have also adopted features for routine use. 

I want people to feel comfortable. I've observed people learning that some are listening "from the lobby". I've considered optional template introduction comments like "Hi, I'm listening from Seattle." I've interviewed newbies about the features, and asked about their perceptions and feelings. I've concluded that a few feel uncomfortable initially, but once they understand the features, some have adopted them. Noticing this, I give demos.

I believe the features contribute to community growth. I'm more connected to the community now that I listen across a number of devices as I exercise, cook, or take a bath. Last night I cooked while Kata and small crew of jazz geniuses played. They're in Europe, and realistically I'm only going to be a listener (and chatter). So what's critical about the fact I'm listening through a browser? 

I originally thought streaming could help outsiders understand what Jamulus is about, and possibly take steps to join us. Maybe that's possible, but moreso I see people who are already in our community just using another way to stay involved on more devices. That's what I'm doing. A guy gave his friend a link today so the outsider could listen in.

https://jamulus.live refreshed 46 browsers at once yesterday. I only measure these simultaneous browser peaks, but suspect hundreds use the service regularly. I continue working to bring new features to the community. I continue trimming features to minimize annoyance factors. I continue listening to feedback. Times change. New ideas develop. And change always evokes feelings. But I see a disconnect between adoption and claims that the sky is falling. I use these features because I'm a die-hard Jamulus fan, and I've heard many express enthusiasm about the new options. So I do not think it's as clear as you claim. But again, it's sensible and fine if you wish to opt out.

**Reply by ElectricBoogalo** [reply]
Thank you John, I believe that is a very reasonable reply, and will help to clarify your position on things. I thought you would be more then willing to accommodate specific use cases, and I'm glad to see that's the case. Everyone uses Jamulus for their own reasons, and you have done a lot to try and make the process easy for end users. Of course some users want to feel a little less "exposed" when in a room, which is also understandable. That we can all appreciate. Cheers.

**Reply by jcritica** [reply]
@mcfnord I couldn"t find another means of contacting you regarding your site so I'll ask here. I noticed that your site doesn't display all occupied servers. I use https://explorer.jamulus.io to view who is online, but that has a per-genre limitation on how things are displayed. I saw references to your site and started looking at it. Having a single page display of servers in use and who's online seems like a good idea, except that what your site displays and who's actually online differs. Most notably during my checks over the last couple weeks servers named CBVB are missing from your site.

Is this a bug on your site?

Thanks

**Reply by jcritica** [reply]
@mcfnord Do you have another contact point regarding your site, and can you address my initial question? Thanks!

**Reply by ElectricBoogalo** [reply]
> @mcfnord Do you have another contact point regarding your site, and can you address my initial question? Thanks!

He has it in his profile: (206) 395-8494 email: sdx.x.jrd@xoxy.net

**Reply by mcfnord** [reply]
Hi @jcritica 

https://jamulus.live does not display all occupied servers. I think there are four scenarios where an occupied server does not appear:

1. It's just occupied by a known bot.
2. The connections on it have been there for more than 6 hours. (If one more connection appears, the site will re-appear.) Sometimes people just park somewhere, and I'm speculating that they aren't really active.
3. The server is only used for a non-music purpose. (For example, there's a group in Canada that speaks Chinese together, with no music.)
4. The name contains "CBVB". These servers routinely IP-ban users. I don't want my users to experience IP-bans, because I worry they'll blame the Jamulus software, so I hide these servers. (Tip: You can still see musicians on those servers using the Musicians tab.)

**Reply by mcfnord** [reply]
@ElectricBoogalo  Based on feedback, I withdrew the recording feature, and only provide streaming at this time. But some users loved recording their jams, so I plan to return a recording option with better labelling (the REC icon) and stricter access rules (can only record on a server you are on).

**Reply by ann0see** [reply]
Probably it would also be beneficial if the bot which connects would then clearly state that it does recording (send message on server,…). Also Server admins should be able to somehow opt out

**Reply by pljones** [reply]
> Also Server admins should be able to somehow opt out

Maybe the bot could look for a welcome message containing a specific exclusion clause.  The bot could also state what it does, how a server admin can opt out and where to find further information.

**Reply by rdica** [reply]
> > Also Server admins should be able to somehow opt out
> 
> Maybe the bot could look for a welcome message containing a specific exclusion clause. The bot could also state what it does, how a server admin can opt out and where to find further information.

+1
Exactly this.

**Reply by mcfnord** [reply]
I have harvested Welcome messages in the past (for Jitsi links), and could support an "opt out keyword" there, but for now I've [added a **Block bots** tab](https://jamulus.live/) with opt-out instructions. Either (a) include "priv" in the server name, or (b) block one or both IPs at the firewall. I'm still intrigued by self-serve opt-out at the individual level as perhaps a better solution. I am also developing a way to provide an informative message in the chat, especially when a recording lobby joins. I reach users in English, German, Chinese, and Thai, which makes disclosures more complicated.

**Reply by ann0see** [reply]
Hmm. I mean probably that solution is fine. Per user level opt out sounds more challenging.

**Reply by rdica** [reply]
@mcfnord Why should **WE** server operators have to change our server names to opt out?

We have branding to consider.
Adding "priv" to our server names would send **OUR** users the wrong message, not to mention new users potentially joining our servers.

Same applies for any thought of per-user opt out using a change in username.

Parsing the server welcome message for a keyword would likely be the best method, much as robots.txt would function.
It doesn't affect server name or username branding, and doesn't require additional firewalling (not every server operator is a server admin-level skilled person), and those that run servers from their home connections (windows and macs mostly) would be able to easily modify welcome message to opt out.

Win-win for everybody.

**Reply by pcar75** [reply]
Keyword in welcome message could be "BOT-FREE" or something similar in english only to reduce the amount of translation guessing or processing required.

**Reply by pljones** [reply]
The other benefit with the welcome message is that it supports some HTML features.  Ideally it could contain, for example:
```
<!--
X-NoBots: all
X-NoBots: broadcast
X-NoBots: broadcast; record; name="Ear"
-->
```
and not be rendered in the actual message...  I wonder if the Qt rich text widget supports it, though - it can be fussy.

(It'll work with the PR for fixing #3050 though.  No client change needed; plain text editing of the server welcome message - or just carry on using `-w '<!-- some comment --><b>LOUD HTML</b>'` command line.  I'm assuming it's _sent_ to the client, regardless of the widget it gets rendered into by the client.)

**Comment by jcritica** [comment]
Here are 3 IP addresses discovered that jamulus.live bots use to record session audio:

`159.89.140.136` (DigitalOcean US West Coast)
`54.67.56.73` (AWS US West)
`134.122.89.41` (DigitalOcean Germany)

**Reply by ElectricBoogalo** [reply]
Thank you! We are in the midst of creating some new servers, and will need to block these programmed listeners IP's to ensure all our jammers are comfortable. Cheers!
